pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -527,19 +552,40 @@ func (p *Planner) updateRepresentationsVariable(fieldConfig *plan.FieldConfigura
 		return
 	}
 
+	if len(p.representationsJson) == 0 {
+		isAbstract := false
+		switch p.visitor.Walker.LastFieldTypeDefinition.Kind {
```

Comment:
I don't know if you need this logic anywhere else, but it may make sense to have an `IsAbstract` method on `Kind`.